[OKTA-459397] Okta Verify (Push/OTP) integration guide (.NET SDK) (#3118)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md

### You're ready to start integrating

> After you've configured your org, you can start integrating the enrollment and challenge flows into your app. The following are the list of supported flows:

Reviewer: ```suggestion
After you configure your org, you can start integrating the enrollment and challenge flows into your app. The following are the list of supported flows:
```

Author: fixed

> * Integrate enrollment using QR Code

Reviewer: ```suggestion
* Integrate enrollment using QR code
```

Author: fixed

* Integrate enrollment using other channels
* Integrate challenge using push notification option
* Integrate challenge using TOTP option

> Each of these has some common steps for initiating the flow once the user has signed in and for polling Okta Verify for a response midway through the flow. We'll cover these first before seeing how they fit into the individual flows

Reviewer: ```suggestion
Each of these has some common steps for initiating the flow after the user is signed in, and for polling Okta Verify for a response midway through the flow. We cover these first before seeing how they fit into the individual flows
```

Author: fixed

## Shared steps between flows

<StackSnippet snippet="commonsteps" />

## Integrate enrollment using QR Code

> In this flow, the user enrols Okta Verify as an authenticator using a QR Code. The following diagram summarizes this flow.

Reviewer: ```suggestion
In this flow, the user enrolls Okta Verify as an authenticator using a QR code. The following diagram summarizes this flow.
```

Author: fixed

<StackSnippet snippet="enrollmentqrcodeintegrationsummary" />

<StackSnippet snippet="enrollmentqrcodeintegrationsteps" />

## Integrate enrollment using other channels

> A user can enrol Okta Verify using several methods other than a QR code. The flow for each of these is the same but we’ll use email as the option in this guide. The following diagram summarizes this flow.

Reviewer: ```suggestion
A user can enroll Okta Verify using several methods other than a QR code. The flow for each of these is the same but we use email as the option in this guide. The following diagram summarizes this flow.
```

<StackSnippet snippet="enrollmentotherintegrationsummary" />

<StackSnippet snippet="enrollmentotherintegrationsteps" />

## Integrate challenge using push notification option

> Once Okta Verify has been enrolled, a user can authenticate themselves with Okta Verify using a push notification challenge. Once they want to log in, they must go to Okta Verify where they must tap on a “Yes its me” prompt sent by the Okta servers. The following diagram summarizes this flow.

Reviewer: ```suggestion
After a user enrolls in Okta Verify, the user can authenticate themselves with Okta Verify using a push notification challenge. When they want to sign in, they must go to Okta Verify where they tap **Yes it's me** sent by the Okta servers. The following diagram summarizes this flow.
```

Author: fixed

<StackSnippet snippet="challengepushintegrationsummary" />

<StackSnippet snippet="challengepushintegrationsteps" />

## Integrate challenge using TOTP option

> Once Okta Verify has been enrolled, a user can authenticate themselves with Okta Verify using a Time-base One-Time Password (TOTP) challenge. Once they want to log in, they must go to Okta Verify which generates the one-time password. The user submits this to the service provider for verification. The Okta server independently generates the password and validates that the submitted password is identical to the generated one in which case the user is authenticated and logged in.

Reviewer: ```suggestion
After a user enrolls in Okta Verify, the user can authenticate themselves with Okta Verify using a Time-based One-Time Password (TOTP) challenge. When they want to sign in, they must go to Okta Verify that generates the TOTP. The user submits this to the service provider for verification. The Okta server independently generates the password and validates that the submitted password is identical to the generated one in, and then the user is authenticated and signed in.
```

Author: fixed
